services/QuillLMS/client/app/bundles/Staff/components/evidence/semanticRules/__tests__/enableMoreThanTenLabelsModal.test.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import EnableMoreThanLabelsModal from '../enableMoreThanTenLabelsModal';
+
+describe('EnableMoreThanLabelsModal', () => {
+  const mockSave = jest.fn();
+  const mockCancel = jest.fn();
+
+  const setup = (isOpen = true) => render(
+    <EnableMoreThanLabelsModal cancel={mockCancel} isOpen={isOpen} save={mockSave} />
+  );
+
+  beforeEach(() => {
+    mockSave.mockClear();
+    mockCancel.mockClear();
+  });
+
+  it('matches the snapshot when modal is open', () => {
+    const { asFragment } = setup()
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('does not render when isOpen is false', () => {
+    setup(false);
+    expect(screen.queryByText(/enable more than 10 labels/i)).not.toBeInTheDocument();
+  });
+
+  it('renders correctly when isOpen is true', () => {
+    setup();
+    expect(screen.getByText(/enable more than 10 labels/i)).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /cancel/i })).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /save/i })).toBeInTheDocument();
+  });
+
+  it('updates input value on change', () => {
+    setup();
+    const input = screen.getByLabelText(/additional labels/i) as HTMLInputElement;
+    fireEvent.change(input, { target: { value: 'label1, label2' } });
```

Comment:
Soft preference for `userEvent.type` here over `fireEvent.change`, just because it more closely simulates the actual interaction (more here: https://testing-library.com/docs/user-event/intro/#:~:text=fireEvent%20dispatches%20DOM%20events%2C%20whereas,additional%20checks%20along%20the%20way.), but not a big deal.

Reply:
Cool.   I like the semantics of `user.type(input, 'label1, label2');` much better.